### PR TITLE
EWPP-671: Add multiselect filter field plugins.

### PR DESCRIPTION
**Pull request by imanoleguskiza**
### Description

Add multiselectfilterfieldplugins

### Change log

- Added: Add multiselectfilterfieldplugins.
- Changed:
- Deprecated:
- Removed:
- Fixed:
- Security:

### Commands

```sh
[Insert commands here]

```

### CODE REVIEW

File: src/Plugin/MultiselectFilterField/BooleanField.php [hunk 1]
```diff
@@ -0,0 +1,73 @@
...
+ */
+class BooleanField extends MultiSelectFilterFieldPluginBase {
+
+  use StringTranslationTrait;
+
+  /**
+   * {@inheritdoc}
+   */
+  public function buildDefaultValueForm(): array {
+    $facet = $this->configuration['facet'];
```
Comment (upchuk) [inline src/Plugin/MultiselectFilterField/BooleanField.php:33]: Don't we expect always the facet to be there in all plugins? Throw an exception in the constructor if the facet is missing and then we can rely on it always being there.

**Review by upchuk — CHANGES_REQUESTED** [review]

File: src/Plugin/facets/widget/ListPagesWidgetBase.php [hunk 2]
```diff
@@ -42,26 +43,8 @@ public function prepareDefaultFilterValue(FacetInterface $facet, array $form, Fo
   /**
    * {@inheritdoc}
    */
-  public function getDefaultValuesLabel(FacetInterface $facet, ListSourceInterface $list_source, ListPresetFilter $filter): string {
```
[the remark this answers is not in the document]
Comment (imanoleguskiza) [reply src/Plugin/facets/widget/ListPagesWidgetBase.php:45]: Fixed.

File: src/ListPresetFiltersBuilder.php [hunk 3]
```diff
@@ -131,7 +131,7 @@ protected function buildSummaryPresetFilters(array $form, FormStateInterface $fo
       $widget = $facet->getWidgetInstance();
       $filter_value_label = '';
       if ($widget instanceof ListPagesWidgetInterface) {
-        $filter_value_label = $widget->getDefaultValuesLabel($facet, $list_source, $filter);
+        $filter_value_label = $widget->getDefaultValuesLabel($facet, $filter, $list_source);
```
[the remark this answers is not in the document]
Comment (imanoleguskiza) [reply src/ListPresetFiltersBuilder.php]: Fixed.

File: .gitignore [hunk 4]
```diff
@@ -12,3 +12,4 @@ phpunit.xml
 runner.yml
 vendor
 yarn.lock
+/.gitattributes
```
[the remark this answers is not in the document]
Comment (imanoleguskiza) [reply .gitignore:15]: This is due to a change in the scaffolding process in the latest drupal versions, the .gitattributes file is placed at the project root so we will need to add it to our .gitignores unless we want to add them to the repo.

**Review by upchuk — APPROVED** [review]

**Review by hernani — APPROVED** [review]

**Pull request merged**